extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -94,6 +127,30 @@ void transformWithSpan(
         }));
     }
 
+    @BuildStep
+    void transformSpanAttribute(
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<AnnotationsTransformerBuildItem> annotationsTransformer) {
+        // Transform deprecated annotation into new one
+        Collection<AnnotationInstance> legacySpanAttribute = combinedIndex.getIndex().getAnnotations(DotName.createSimple(
+                "io.opentelemetry.extension.annotations.SpanAttribute"));
+
+        // Transform deprecated annotation into new one
+        annotationsTransformer.produce((new AnnotationsTransformerBuildItem(context -> {
+            AnnotationTarget target = context.getTarget();
+            for (AnnotationInstance legacyAnnotation : legacySpanAttribute) {
+                if (target.equals(legacyAnnotation.target())) {
+                    AnnotationValue value = legacyAnnotation.value("value");
+                    AnnotationInstance annotation = AnnotationInstance.create(
+                            DotName.createSimple("io.opentelemetry.instrumentation.annotations.SpanAttribute"),
+                            target,
+                            List.of(value));
+                    context.transform().add(annotation).done();
+                }
+            }
+        })));
+    }
```

Comment:
This transformation is useless I guess? `@SpanAttribute` is not a CDI-related annotation, and the runtime code below looks for the legacy annotation anyway.

Reply:
yes, unfortunately the interceptor has no other way but look for the deprecated annotation as well. This ended up not being needed. Will remove.